client/my-sites/controller.js

```diff
@@ -183,6 +185,18 @@ module.exports = {
 				// if sites have loaded, but siteID is invalid, redirect to allSitesPath
 				if ( sites.select( siteID ) ) {
 					onSelectedSiteAvailable();
+				} else if ( ( currentUser.visible_site_count !== sites.getVisible().length ) ) {
+					ReactDom.render( (
+						<Main>
+							<JetpackManageErrorPage template="waitForJetpackSite" />
+						</Main>
+					), document.getElementById( 'primary' ) );
+
+					setTimeout( () => {
```

Comment:
I'd model it on the `sites.once` call here: https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/controller.js#L175-L190

- You check if site_count is different from sites.length.
- If so you send a sites fetch and listen to the change event.
- Run the comparison again to know you got the site.